.github/workflows/run-test-suite.yml

```diff
@@ -0,0 +1,22 @@
+name: run-test-suite
+on: [push, pull_request]
```

Comment:
@matthieucan I changed the workflow to trigger on pull requests as well, and you can see it in action in this very pull request!

The workflow will not run if there's a merge conflict. It will also not run if you're a first-time contributor, but we can still manually approve a workflow run for those PRs.